src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.SemanticChangeProcessor.cs

```diff
@@ -48,6 +48,20 @@ private sealed class SemanticChangeProcessor : IdleProcessor
                     _pendingWork = new Dictionary<DocumentId, Data>();
 
                     Start();
+
+                    AsyncProcessorTask.ContinueWith(
+                        _ =>
+                        {
+                            foreach (var (documentId, data) in _pendingWork)
+                            {
+                                data.AsyncToken.Dispose();
+                            }
+
+                            _pendingWork.Clear();
```

Comment:
>_pendingWork.Clear(); [](start = 28, length = 21)

Why aren't the operations on `_pendingWork` guarded by the appropriate gate here?

Reply:
These operations are only executed after the `_pendingWork` processing method completes. There isn't a way for it to be accessed at the same time.

Reply:
Fixed in 9c28f4a